tests/unit/module_utils/elbv2/test_listener_rules.py

```diff
@@ -0,0 +1,375 @@
+        ],
+        {
+            "to_add": [
+                {
+                    "Priority": 2,
+                    "Conditions": [{"Field": "host-header", "Values": ["yolo.rocks"]}],
+                    "Actions": [{"TargetGroupName": "target2", "Type": "forward"}],
+                },
+            ]
+        },
+    ),
+    (
+        [
+            {
+                "Priority": "1",
+                "Conditions": [{"Field": "host-header", "Values": ["bla.tld"]}],
+                "Actions": [{"TargetGroupName": "target1", "Type": "forward"}],
+                "RuleArn": "arn:aws:elasticloadbalancing:::listener-rule/app/ansible-test/abc",
+            },
+            {
+                "Priority": "2",
+                "Conditions": [{"Field": "host-header", "Values": ["yolo.rocks"]}],
+                "Actions": [{"TargetGroupName": "target2", "Type": "forward"}],
+                "RuleArn": "arn:aws:elasticloadbalancing:::listener-rule/app/ansible-test/123",
+            },
+        ],
+        [
+            {
+                "Priority": 1,
+                "Conditions": [{"Field": "host-header", "Values": ["bla.tld"]}],
+                "Actions": [{"TargetGroupName": "target1", "Type": "forward"}],
+            },
+            {
+                "Priority": 2,
+                "Conditions": [{"Field": "path-pattern", "Values": ["/test"]}],
+                "Actions": [
+                    {"TargetGroupName": "oidc-target-01", "Type": "forward", "Order": 2},
+                    {
+                        "Type": "authenticate-oidc",
+                        "Order": 1,
+                        "AuthenticateOidcConfig": {
+                            "Issuer": "https://sample.oauth.com/issuer",
+                            "AuthorizationEndpoint": "https://sample.oauth.com",
+                            "TokenEndpoint": "https://sample.oauth.com/oauth/token",
+                            "UserInfoEndpoint": "https://sample.oauth.com/userinfo",
+                            "ClientId": "id123645",
+                            "ClientSecret": "testSecret123!@#$",
+                            "UseExistingClientSecret": True,
+                        },
+                    },
+                ],
+            },
+            {
+                "Priority": 3,
+                "Conditions": [{"Field": "host-header", "Values": ["yolo.rocks"]}],
+                "Actions": [{"TargetGroupName": "target2", "Type": "forward"}],
+            },
+        ],
+        {
+            "to_modify": [
+                {
+                    "Priority": 2,
+                    "Conditions": [{"Field": "path-pattern", "Values": ["/test"]}],
+                    "Actions": [
+                        {"TargetGroupName": "oidc-target-01", "Type": "forward", "Order": 2},
+                        {
+                            "Type": "authenticate-oidc",
+                            "Order": 1,
+                            "AuthenticateOidcConfig": {
```

Comment:
Could we also add a test case like this but where the current listener includes an authenticate-oidc action and the client secret is removed in the listener to_modify? I just want to make sure both cases from the original issue are represented.

Reply:
I have added new test cases for the authenticated-oidc action